File: backend/crawl/app/models/rank.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.models.database import Base


class Rank(Base):
    __tablename__ = "rank"

    id = Column(Integer, primary_key=True, index=True)
    year = Column(String(10))
    rank = Column(Integer)
    team_id = Column(Integer, ForeignKey("team.id"))
    game_count = Column(Integer)
    win_count = Column(Integer)
    lose_count = Column(Integer)
    draw_count = Column(Integer)
    win_rate = Column(String(10))
    game_diff = Column(String(10))
    recent_10 = Column(String(20))
    streak = Column(String(10))
    rank_change = Column(Integer)
    
    team = relationship("Team", back_populates="rank")

File: backend/baseball/src/main/java/com/newbie/baseball/domain/rank/entity/Rank.java
package com.newbie.baseball.domain.rank.entity;

import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "`rank`")
public class Rank {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "year", length = 10)
    private String year;

    @Column(name = "`rank`")
    private Integer rank;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Column(name = "game_count")
    private Integer gameCount;

    @Column(name = "win_count")
    private Integer winCount;

    @Column(name = "lose_count")
    private Integer loseCount;

    @Column(name = "draw_count")
    private Integer drawCount;

    @Column(name = "win_rate", length = 10)
    private String winRate;

    @Column(name = "game_diff", length = 10)
    private String gameDiff;

    @Column(name = "recent_10", length = 20)
    private String recent10;

    @Column(name = "streak", length = 10)
    private String streak;

    @Column(name = "rank_change")
    private Integer rankChange;
}